qcfractal/testing.py

```diff
@@ -533,10 +536,73 @@ def sqlalchemy_socket_fixture(request, postgres_server):
 
 
 def live_fractal_or_skip():
-    """Ensure Fractal live connection can be made"""
+    """
+    Ensure Fractal live connection can be made
+    First looks for a local staging server, then tries QCArchive.
+    """
     try:
-        import qcfractal.interface
-        requests.get('https://api.qcarchive.molssi.org:443', json={}, timeout=5)
-        return FractalClient()
+        return FractalClient('localhost:7777', verify=False)
     except (requests.exceptions.ConnectionError, ConnectionRefusedError):
-        return pytest.skip("Could not make a connection to central Fractal server")
+        print("Failed to connect to localhost")
+        try:
+            return pytest.xfail("Until QCA is migrated to v0.12.0")
+            requests.get('https://api.qcarchive.molssi.org:443', json={}, timeout=5)
+            return FractalClient()
+        except (requests.exceptions.ConnectionError, ConnectionRefusedError):
+            return pytest.skip("Could not make a connection to central Fractal server")
+
+
+def df_equals(df1, df2, sort=False):
+    """ checks equality even when columns contain numpy arrays, which .equals and == struggle with """
+    if sort:
+        if isinstance(df1, pd.DataFrame):
+            df1 = df1.reindex(sorted(df1.columns), axis=1)
+        elif isinstance(df1, pd.Series):
+            df1 = df1.sort_index()
+        if isinstance(df2, pd.DataFrame):
+            df2 = df2.reindex(sorted(df2.columns), axis=1)
+        elif isinstance(df2, pd.Series):
+            df2 = df2.sort_index()
+
+    def element_equal(e1, e2):
+        if isinstance(e1, np.ndarray):
+            if not np.array_equal(e1, e2):
+                return False
+        elif isinstance(e1, Molecule):
+            if not e1.get_hash() == e2.get_hash():
```

Comment:
We should build this into the Molecule class.

Reply:
Agreed. Resolved because we have an issue in elemental now.